### DISCUSSION TITLE
Bootstrap node implementation

### DISCUSSION THREAD

**Discussion by t-botclive** [libp2p/rust-libp2p#5353] [Q&A]
Im looking for implementing a bootstrap node, and I want it to store only peers related to my network, so it wont become a public ipfs node with tons of different peers. Im using kademlia. Since there is no `namespace` feature like in go-libp2p, is there a way I can configure swarm or is there any suggested playaround to achieve this? Im also using gossipsub for sending messages and topic can be used as a validation criteria, but in this case boot node will receive messages from other nodes which is not good.

**Comment by dariusc93** [comment]
You could change the protocol name, which *should* allow you to do what youre seeking to do. See [Config::new](https://github.com/libp2p/rust-libp2p/blob/master/protocols/kad/src/behaviour.rs#L218) and [ConfigBuilder::protocol_id](https://github.com/libp2p/rust-libp2p/blob/master/protocols/gossipsub/src/config.rs#L468). Another option would be to have a protocol added to your peers (it can be NOOP) and check their protocols after connecting (either in the connection handler, via identify or any future behaviour that reports the remote protocols) then store them accordingly if you have your own db or behaviour storing peers info (though dont think this is the case)
